linera-views/src/map_view.rs

```diff
@@ -377,22 +480,70 @@ where
     ///   map.insert(vec![0,1], String::from("Hello"));
     ///   map.insert(vec![1,2], String::from("Bonjour"));
     ///   map.insert(vec![2,2], String::from("Hallo"));
+    ///   assert_eq!(map.count().await.unwrap(), 3);
+    /// # })
+    /// ```
+    pub async fn count(&self) -> Result<usize, ViewError> {
+        let mut count = 0;
+        let prefix = Vec::new();
+        self.for_each_key(
+            |_key| {
+                count += 1;
+                Ok(())
+            },
+            prefix,
+        )
+        .await?;
+        Ok(count)
+    }
+
+    /// Applies a function f on each index/value pair matching a prefix. Keys
+    /// and values are visited in the lexicographic order. The shortened index
+    /// is send to the function f and if it returns false then the loop ends
+    /// prematurely
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::memory::create_memory_context;
+    /// # use linera_views::map_view::ByteMapView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_memory_context();
+    ///   let mut map = ByteMapView::load(context).await.unwrap();
+    ///   map.insert(vec![0,1], String::from("Hello"));
+    ///   map.insert(vec![1,2], String::from("Bonjour"));
+    ///   map.insert(vec![1,3], String::from("Hallo"));
     ///   let mut part_keys = Vec::new();
+    ///   let prefix = vec![1];
     ///   map.for_each_key_value_while(|key, _value| {
     ///     part_keys.push(key.to_vec());
     ///     Ok(part_keys.len() < 2)
-    ///   }).await.unwrap();
+    ///   }, prefix).await.unwrap();
     ///   assert_eq!(part_keys.len(), 2);
     /// # })
     /// ```
-    pub async fn for_each_key_value_while<F>(&self, mut f: F) -> Result<(), ViewError>
+    pub async fn for_each_key_value_while<F>(
```

Comment:
(The same comments as above apply to this function.)

Reply:
Of course.